app/containers/ItunesProvider/ItunesGridContainer/index.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import debounce from 'lodash/debounce';
+import get from 'lodash/get';
+import { Card, Input } from 'antd';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const { Search } = Input;
+
+const Songs = styled.div`
+  max-width: 2 * ${(props) => props.maxwidth}px;
+  padding: ${(props) => props.padding}px;
+  margin: auto;
+`;
+
+const SearchList = styled.div``;
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px auto;
+    max-width: ${(props) => props.maxwidth}px;
+    padding: ${(props) => props.padding}px;
+  }
+`;
+
+const ItunesGridContainer = ({ maxwidth, padding, dispatchSongName, songs }) => {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(songs, 'results', null);
+    if (loading && loaded) {
+      setLoading(false);
+    }
+  }, [songs]);
+
+  const handleOnChange = (sName) => {
+    if (sName) {
+      dispatchSongName(sName);
+      setLoading(true);
+    }
```

Comment:
in this case when the `sName` doesn't have a value it will hold the last value instead. Create a clear tracks option and dispatch that in this case.